packages/material-ui/src/Modal/Modal.test.js

```diff
@@ -382,193 +363,198 @@ describe('<Modal />', () => {
-      wrapper = mount(
+      render(
         <Modal open disableAutoFocus>
           <div>Foo</div>
         </Modal>,
       );
-      assert.strictEqual(document.activeElement, initialFocus);
-    });
 
-    it('should not focus modal when child has focus', () => {
-      wrapper = mount(
-        <Modal open>
-          <div>
-            <input autoFocus />
-          </div>
-        </Modal>,
-      );
-      assert.strictEqual(document.activeElement, document.querySelector('input'));
+      expect(initialFocus).toHaveFocus();
     });
 
     it('should return focus to the modal', () => {
-      wrapper = mount(
+      const { getByTestId } = render(
         <Modal open>
-          <div className="modal">
-            <input autoFocus />
+          <div data-testid="modal">
+            <input autoFocus data-testid="auto-focus" />
           </div>
         </Modal>,
       );
 
-      assert.strictEqual(document.activeElement, document.querySelector('input'));
+      expect(getByTestId('auto-focus')).toHaveFocus();
+
       initialFocus.focus();
-      assert.strictEqual(document.activeElement.className, 'modal');
+
+      expect(getByTestId('modal')).toHaveFocus();
     });
 
     it('should not return focus to the modal when disableEnforceFocus is true', () => {
-      wrapper = mount(
+      const { getByTestId } = render(
         <Modal open disableEnforceFocus>
-          <div className="modal">
-            <input autoFocus />
+          <div>
+            <input autoFocus data-testid="auto-focus" />
           </div>
         </Modal>,
       );
 
-      assert.strictEqual(document.activeElement, document.querySelector('input'));
+      expect(getByTestId('auto-focus')).toHaveFocus();
+
       initialFocus.focus();
-      assert.strictEqual(document.activeElement, initialFocus);
+
+      expect(initialFocus).toHaveFocus();
     });
 
     it('should warn if the modal content is not focusable', () => {
-      const Dialog = React.forwardRef((_, ref) => <div ref={ref} />);
+      const UnfocusableDialog = React.forwardRef((_, ref) => <div ref={ref} />);
 
-      wrapper = mount(
+      render(
         <Modal open>
-          <Dialog />
+          <UnfocusableDialog />
         </Modal>,
       );
-      assert.strictEqual(consoleErrorMock.callCount(), 1);
-      assert.match(consoleErrorMock.messages()[0], /the modal content node does not accept focus/);
+      expect(consoleErrorMock.callCount()).to.equal(1);
+      expect(consoleErrorMock.messages()[0]).to.include(
+        'the modal content node does not accept focus',
+      );
     });
 
     it('should not attempt to focus nonexistent children', () => {
-      const Dialog = () => null;
+      const EmptyDialog = () => null;
 
-      wrapper = mount(
+      render(
         <Modal open>
-          <Dialog />
+          <EmptyDialog />
         </Modal>,
       );
     });
 
     it('should loop the tab key', () => {
```

Comment:
Wow, I'm blown away, I thought we couldn't test the tab behavior https://github.com/testing-library/react-testing-library/issues/376. Somehow I did in #14545. I'm surprised, I didn't remember.

Regarding the test regression, replacing testing-library fire event with

```jsx
      const dispatchKey = (properties) => {
        const event = new window.Event('keydown');
        Object.keys(properties).forEach((key) => {
          event[key] = properties[key];
        });
        document.dispatchEvent(event);
      };
```

solves the fail.

Reply:
> Wow, I'm blown away, I thought we couldn't test the tab behavior 

we can't. It's not actual tab behavior. We assume in `TrapFocus` that there is any and intercept it.

I'll try your fix. I think this is something we want to get in `@testing-library/dom` regardless.

Reply:
Eww, I thought that Cypress has support for it: https://docs.cypress.io/api/commands/type.html#Tabbing.

Reply:
Oh ok, I figured it out, tab focus definitely doesn't work.

Reply:
TrapFocus is responsible for calling `.focus()`, the issue with the new test is that `event.keyCode` has the default value.